examples/authentication.test.js

```diff
@@ -1,30 +1,31 @@
-import createJWKSMock from '../src/index'
-import createApp from './api'
+// @ts-check
+import { createJWKSMock } from 'mock-jwks'
+import { createApp } from './api.js'
 import supertest from 'supertest'
-import { beforeEach, describe, expect, test } from 'vitest'
+import { describe, expect, test, onTestFinished } from 'vitest'
 
-describe('Some tests for authentication for our api', () => {
-  let jwksMock, server, request
-  beforeEach(() => {
-    ;({ jwksMock, server, request } = createContext())
-    return () => tearDown({ jwksMock, server })
-  })
+// This creates the local PKI
+const jwksMock = createJWKSMock('https://levino.eu.auth0.com')
+const app = createApp({
+  jwksUri: 'https://levino.eu.auth0.com/.well-known/jwks.json',
+})
 
+describe('Some tests for authentication for our api', () => {
   test('should not get access without correct token', async () => {
     // We start intercepting queries (see below)
-    jwksMock.start()
-    const { status } = await request.get('/')
+    onTestFinished(jwksMock.start())
+    const { status } = await supertest(app).get('/')
```

Comment:
Since we are now using an express app, there is no need to start it by hand for supertest. Way better.